src/main/java/frc/robot/vision/sources/RobotPoseEstimatingVisionSource.java

```diff
@@ -0,0 +1,17 @@
+package frc.robot.vision.sources;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.vision.GyroAngleValues;
+import frc.robot.vision.RawVisionAprilTagData;
+
+import java.util.Optional;
+
+public interface RobotPoseEstimatingVisionSource extends VisionSource {
+
+	void updateGyroAngles(GyroAngleValues gyroAngleValues);
+
+	Optional<Rotation2d> getRobotHeading();
+
+	Optional<RawVisionAprilTagData> getRawVisionData();
```

Comment:
I think it's should be moved to VisionSource to help generalize MultiVisionSources (I have a draft in my stash, where VisionSource getRawVisionData at VisionSource returns a generic and in RobotPoseEstimatingVisionSource it's non-generic).